xline/src/storage/kvwatcher.rs

```diff
@@ -262,25 +359,81 @@ where
         storage: Arc<KvStore<S>>,
         mut kv_update_rx: mpsc::Receiver<(i64, Vec<Event>)>,
     ) -> Arc<Self> {
+        let watcher_map = Arc::new(RwLock::new(WatcherMap::new()));
         let kv_watcher = Arc::new(Self {
             storage,
-            watcher_map: RwLock::new(WatcherMap::new()),
+            watcher_map,
         });
-        let watcher = Arc::clone(&kv_watcher);
-        let _handle = tokio::spawn(async move {
-            while let Some(updates) = kv_update_rx.recv().await {
-                watcher.handle_kv_updates(updates);
+        let _victim_handle = tokio::spawn(Arc::clone(&kv_watcher).sync_victims_task());
+        let _kv_updates_handle = tokio::spawn({
+            let kv_watcher = Arc::clone(&kv_watcher);
+            async move {
+                while let Some(updates) = kv_update_rx.recv().await {
+                    kv_watcher.handle_kv_updates(updates);
+                }
             }
         });
         kv_watcher
     }
 
+    /// Background task to sync victims
+    async fn sync_victims_task(self: Arc<Self>) {
```

Comment:
Shall we put this background task under the control of shutdown_trigger like some other background tasks does?